src/main/java/at/hannibal2/skyhanni/data/QuiverAPI.kt

```diff
@@ -0,0 +1,238 @@
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.NumberUtil.formatNumber
+import at.hannibal2.skyhanni.utils.SkyBlockItemModifierUtils.getEnchantments
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.StringUtils.removeResets
+import at.hannibal2.skyhanni.utils.StringUtils.trimWhiteSpace
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.client.Minecraft
+import net.minecraft.item.ItemBow
+import net.minecraftforge.fml.common.eventhandler.EventPriority
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+private var infinityQuiverLevelMultiplier = 0.03f
+
+object QuiverAPI {
+    private val storage get() = ProfileStorageData.profileSpecific
+    var currentArrow: ArrowType?
+        get() =  storage?.arrows?.currentArrow?.asInternalName()?.let{ getArrowByNameOrNull(it) } ?: NONE_ARROW_TYPE
+        set(value) {
+            storage?.arrows?.currentArrow = value?.toString() ?: return
+        }
+    var arrowAmount: MutableMap<NEUInternalName, Float>
+        get() = storage?.arrows?.arrowAmount ?: mutableMapOf()
+        set(value) {
+            storage?.arrows?.arrowAmount = value
+        }
+    var currentAmount: Int
+        get() = arrowAmount[currentArrow?.internalName]?.toInt() ?: 0
+        set(value) {
+            arrowAmount[currentArrow?.internalName ?: return] = value.toFloat()
+        }
+
+    private var arrows: List<ArrowType> = listOf()
+
+    const val MAX_ARROW_AMOUNT = 2880
+    private val SKELETON_MASTER_CHESTPLATE = "SKELETON_MASTER_CHESTPLATE".asInternalName()
+
+    private var NONE_ARROW_TYPE: ArrowType? = null
+    private var FLINT_ARROW_TYPE: ArrowType? = null
+
+    private val group = RepoPattern.group("data.quiver")
+    private val chatGroup = group.group("chat")
+    private val selectPattern by chatGroup.pattern("select", "§aYou set your selected arrow type to §f(?<arrow>.*)§a!")
+    private val fillUpJaxPattern by chatGroup.pattern(
+        "fillupjax",
+        "(§.)*Jax forged (§.)*(?<type>.*?)(§.)* x(?<amount>[\\d,]+)( (§.)*for (§.)*(?<coins>[\\d,]+) Coins)?(§.)*!"
+    )
+    private val fillUpPattern by chatGroup.pattern(
+        "fillup",
+        "§aYou filled your quiver with §f(?<flintAmount>.*) §aextra arrows!"
+    )
+    private val clearedPattern by chatGroup.pattern("cleared", "§aCleared your quiver!")
+    private val arrowResetPattern by chatGroup.pattern("arrowreset", "§cYour favorite arrow has been reset!")
+    private val addedToQuiverPattern by chatGroup.pattern(
+        "addedtoquiver",
+        "(§.)*You've added (§.)*(?<type>.*) x(?<amount>.*) (§.)*to your quiver!"
+    )
+
+    // Bows that don't use the players arrows
+    private val fakeBowsPattern by group.pattern("fakebows", "^BOSS_SPIRIT_BOW$")
```

Comment:
![image](https://github.com/hannibal002/SkyHanni/assets/69345714/d077bbf4-d909-4565-bfd9-9408c823f260)

:)

Reply:
> Surely it still uses arrows?

From my testing, no. The stack number of the quiver arrow in the 9th slot goes down, but it doesnt actually consume one.

Reply:
Ah sick thank you

Reply:
Should be [fixed](https://github.com/hannibal002/SkyHanni/commit/442e2f1ab66e1134135c35544186bd6f47e5adc4)